ts/src/coinbase.ts

```diff
@@ -344,6 +344,7 @@ export default class coinbase extends Exchange {
                 'CGLD': 'CELO',
             },
             'options': {
+                'usePrivateEndpointsIfApikeysPresent': false,
```

Comment:
I wouldn't do it like this. This is nonstandard.

We should read an option`method` and maybe `private` using handleOptionAndParams (and add that to the docs). then inside each method we use that to decide if we use one or the other endpoint

Reply:
Take a look at this commit a1fe0b95020254445f5919af7eead84c6b109b2b which includes a test for credentials and switches from public to private API calls ... this is the pattern I was suggesting earlier.  Perhap I'm not reading this correctly?  Decide if this is non-standard approach or not?
+        if (this.checkRequiredCredentials (false)) {
+            spotUnresolvedPromises.push (this.v3PrivateGetBrokerageTransactionSummary (params));
+        }

Reply:
@carlosmiei so do you suggest that we have to add `private: true` in 5 methods under `options` one-by-one?  also `private` is not obviously intuitive, we might need more expressive key name.

Reply:
you don't need to add one by one in options, but we can have a global option like `usePrivateEndpoint` and inside each method you read the value using `handleOptionAndParams`

Reply:
> you don't need to add one by one in options, but we can have a global option like `usePrivateEndpoint` and inside each method you read the value using `handleOptionAndParams`

but is not that what is being done in this PR?

Reply:
kindof, the `useprivate` is adding an unnecessary layer I think, you're not reading from `params`

Reply:
but what should be the meaningful name so users didn't ask "what does this option mean"?

Reply:
i've marked this ready for review, please feel free to edit that name inline as you prefer, otherwise it's ready